### PR TITLE
Add teads bidder adapter

### PR DESCRIPTION
**Pull request by valsouche**
## Type of change

- [ ] Bugfix
- [ ] Feature
- [x] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change

New teads bidder adapter

- test parameters for validating bids
```
{
  bidder: 'teads',
  params: {
    pageId: 89489,
    placementId: 96888
  }
}
```

Be sure to test the integration with your adserver using the [Hello World](/integrationExamples/gpt/hello_world.html) sample page.

- innov-ssp@teads.tv
- [x] official adapter submission

For any changes that affect user-facing APIs or example code documented on http://prebid.org, please provide:

- https://github.com/prebid/prebid.github.io/pull/978

### CODE REVIEW

File: modules/teadsBidAdapter.js [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+   * @return {Bid[]} An array of bids which were nested inside the server.
+   */
+  interpretResponse: function(serverResponse, bidderRequest) {
+    const bidResponses = [];
+    serverResponse = serverResponse.body;
+
+    if (serverResponse.responses) {
+      serverResponse.responses.forEach(function (bid) {
+        const bidResponse = {
+          bidderCode: spec.code,
```
Comment (jsnellbaker) [inline modules/teadsBidAdapter.js]: Can be removed; not needed in the response object.

File: test/spec/modules/teadsBidAdapter_spec.js [hunk 2]
```diff
@@ -0,0 +1,254 @@
+import {expect} from 'chai';
+import {spec} from 'modules/teadsBidAdapter';
+import {newBidder} from 'src/adapters/bidderFactory';
+
+const ENDPOINT = '//a.teads.tv/hb/bid-request';
+const AD_SCRIPT = '<script type="text/javascript" class="teads" async="true" src="http://a.teads.tv/hb/bid-request/hb/getAdSettings"></script>"';
+
+describe('teadsBidAdapter', () => {
```
Comment (jsnellbaker) [inline test/spec/modules/teadsBidAdapter_spec.js]: Can you change the arrow syntax in `describe` functions to use the normal syntax?  Ie change `, () => {` to `, function() {`

The normal syntax is more inline with Mocha, and most other adapter's test files now use this style.

**Comment by valsouche** [comment]
@jsnellbaker , thanks for review.
About the demonstration page and placement ids, you have to use values from pull request description instead of `.md` because these ones are fake.
I've just tested on hello_world and this is ok. Thanks !

**Comment by jsnellbaker** [comment]
Hi @valsouche 

Thanks for the update; however I tried to use the params from the PR description and I'm still seeing the same empty response.

Below is a copy of the request payload when it was executed:
```
{"referrer":"http://ap.localhost:9999/integrationExamples/gpt/hello_world.html?pbjs_debug=true","data":[{"sizes":["300x250","300x600"],"bidId":"2c6a2963ce8209","bidderRequestId":"1fbc5df52cca93","placementId":96888,"pageId":89489,"adUnitCode":"div-gpt-ad-1460505748561-0","auctionId":"371aca91-70b2-4ab1-9fa9-0a3f0dcfcd9d","transactionId":"2541c605-bf5b-4b84-aa6f-f50b3c5aba8c"}],"deviceWidth":1680}
```
Do you see anything amiss in the above?

**Comment by valsouche** [comment]
Hello @jsnellbaker, 
We have checked all our setup, everything looks ok now. We have delivery on each call, with this payload which is very close to yours : 

```
{"referrer":"http://localhost:9999/integrationExamples/gpt/hello_world.html","data":[{"sizes":["300x250","300x600"],"bidId":"2bc41155d9083c","bidderRequestId":"1bb845bca09b36","placementId":96888,"pageId":89489,"adUnitCode":"div-gpt-ad-1460505748561-0","auctionId":"aef26605-6bee-4430-89c0-d3853de17ebe","transactionId":"42aec225-1ba8-4101-94b6-2ab5c28f56d9"}],"deviceWidth":1680}
```

If you still have no delivery after multiple tries, please tell us and we will create a new test setup.

Thanks.

**Pull request merged**